src/org/rascalmpl/library/lang/json/internal/JsonValueWriter.java

```diff
@@ -367,19 +376,25 @@ public Void visitBoolean(IBool boolValue) throws IOException {
 
       @Override
       public Void visitExternal(IExternalValue externalValue) throws IOException {
-        // TODO
         throw new IOException("External values are not supported by JSon serialisation yet");
       }
 
       @Override
       public Void visitDateTime(IDateTime o) throws IOException {
         if (datesAsInts) {
           out.value(o.getInstant());
-          return null;
         }
         else {
-          throw new IOException("Dates as strings not yet implemented: " + format.get().toPattern());
+          try {
+            com.ibm.icu.text.SimpleDateFormat sd = format.get(); 
```

Comment:
can we not use java time for this? since java8 it's been really a nice API, and I would love to reduce our dependency on icu for date time stuff.

Reply:
We could but this semantics has to remain consistent with the `DateTime` module for now. When we move to java time, we move it all. There are so many subtleties in this stuff that I don't want to multiply the confusion.